### PR TITLE
pre/post send/receive command fixes

### PR DESCRIPTION
**Pull request by dunron**
Just a couple of fixes.

### CODE REVIEW

File: lib/ZnapZend.pm [hunk 1]
```diff
@@ -135,10 +135,17 @@ my $refreshBackupPlans = sub {
         for (keys %$backupSet){
             my ($key) = /^dst_([^_]+)_precmd$/ or next;
 
-            #perform destination remote pre-command if any
-            $self->zLog->info("running pre-command for " . $backupSet->{"dst_$key"});
-            system($backupSet->{"dst_$key" . '_precmd'})
-            && $self->zLog->warn("command \'" . $backupSet->{"dst_$key" . '_precmd'} . "\' failed");
+            #perform pre-send-command if any
+            if ($backupSet->{"dst_$key" . '_precmd'} && $backupSet->{"dst_$key" . '_precmd'} ne 'off'){
```
Comment (oetiker) [inline lib/ZnapZend.pm]: there should be documentation on the special value 'off'

File: lib/ZnapZend.pm [hunk 2]
```diff
@@ -184,10 +191,17 @@ my $refreshBackupPlans = sub {
         for (keys %$backupSet){
             my ($key) = /^dst_([^_]+)_pstcmd$/ or next;
 
-            #perform destination remote post-command if any
-            $self->zLog->info("running post-command for " . $backupSet->{"dst_$key"});
-            system($backupSet->{"dst_$key" . '_pstcmd'})
-            && $self->zLog->warn("command \'" . $backupSet->{"dst_$key" . '_pstcmd'} . "\' failed");
+            #perform post-send-command if any
+            if ($backupSet->{"dst_$key" . '_pstcmd'} && $backupSet->{"dst_$key" . '_pstcmd'} ne 'off'){
```
Comment (oetiker) [inline lib/ZnapZend.pm]: off needs documentation

File: lib/ZnapZend.pm [hunk 3]
```diff
@@ -220,11 +234,14 @@ my $sendRecvCleanup = sub {
     for my $dst (sort grep { /^dst_[^_]+$/ } keys %$backupSet){
         my ($key) = $dst =~ /dst_([^_]+)$/;
 
-        #check destination for remote pre-command
-        if ($backupSet->{$dst . '_precmd'}){
-            $self->zLog->info("running pre-command for " . $backupSet->{$dst});
-            system($backupSet->{$dst . '_precmd'})
-            && $self->zLog->warn("command \'" . $backupSet->{$dst . '_precmd'} . "\' failed");
+        #check destination for pre-send-command
+        if ($backupSet->{"dst_$key" . '_precmd'} && $backupSet->{"dst_$key" . '_precmd'} ne 'off'){
```
Comment (oetiker) [inline lib/ZnapZend.pm]: off needs documentation

File: lib/ZnapZend.pm [hunk 4]
```diff
@@ -302,11 +319,14 @@ my $sendRecvCleanup = sub {
             }
         }
 
-        #check destination for remote post-command
-        if ($backupSet->{$dst . '_pstcmd'}){
-            $self->zLog->info("running post-command for " . $backupSet->{$dst});
-            system($backupSet->{$dst . '_pstcmd'})
-            && $self->zLog->warn("command \'" . $backupSet->{$dst . '_pstcmd'} . "\' failed");
+        #check destination for remote post-send-command
+        if ($backupSet->{"dst_$key" . '_pstcmd'} && $backupSet->{"dst_$key" . '_pstcmd'} ne 'off'){
```
Comment (oetiker) [inline lib/ZnapZend.pm]: off needs documentation

File: bin/znapzendzetup [hunk 5]
```diff
@@ -558,6 +558,10 @@ setup into ~root/.my.cnf and make sure the file permissions are tight ...
 
 Specify delay (in seconds) before sending snaps to the destination. May be useful if you want to control sending time.
 
+=item B<pre-send-command post-send-command>
+
+Run command/script before and after sending the snapshot to the destination. Intended to run a remote script via ssh on the destination, e.g. to bring up a backup disk or server. Or to bring a zpool online/offline: "ssh root@bserv zpool import -Nf tank" "ssh root@bserv zpool export tank".
```
Comment (oetiker) [inline bin/znapzendzetup]: use linebreaking

**Comment by oetiker** [comment]
your PR can not be merged because it has conflicts!

**Comment by dunron** [comment]
Sorry about the conflicts. Pls have another look.

**Comment by oetiker** [comment]
thanks!

**Pull request merged**